krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -17,224 +17,41 @@
 package v1alpha1
 
 import (
-	"fmt"
-
 	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
 	nephioreqv1alpha1 "github.com/nephio-project/api/nf_requirements/v1alpha1"
-	"sigs.k8s.io/yaml"
-)
-
-const (
-	// errors
-	errKubeObjectNotInitialized = "KubeObject not initialized"
+	"github.com/nephio-project/nephio/krm-functions/lib/kubeobject"
 )
 
-var (
-	attachmentType      = []string{"spec", "attachmentType"}
-	cniType             = []string{"spec", "cniType"}
-	networkInstanceName = []string{"spec", "networkInstance", "name"}
-)
+type Interface struct {
+	kubeobject.KubeObjectExt[*nephioreqv1alpha1.Interface]
+}
 
-type Interface interface {
-	// GetKubeObject returns the present kubeObject
-	GetKubeObject() *fn.KubeObject
-	// GetGoStruct returns a go struct representing the present KRM resource
-	GetGoStruct() (*nephioreqv1alpha1.Interface, error)
-	// GetAttachmentType returns the attachmentType from the spec
-	// if an error occurs or the attribute is not present an empty string is returned
-	GetAttachmentType() string
-	// GetCNIType returns the cniType from the spec
-	// if an error occurs or the attribute is not present an empty string is returned
-	GetCNIType() string
-	// GetNetworkInstanceName returns the name of the networkInstance from the spec
-	// if an error occurs or the attribute is not present an empty string is returned
-	GetNetworkInstanceName() string
-	// SetAttachmentType sets the attachmentType in the spec
-	SetAttachmentType(s string) error
-	// SetCNIType sets the cniType in the spec
-	SetCNIType(s string) error
-	// SetNetworkInstanceName sets the name of the networkInstance in the spec
-	SetNetworkInstanceName(s string) error
-	// SetSpec sets the spec attributes in the kubeobject according the go struct
-	SetSpec(*nephioreqv1alpha1.InterfaceSpec) error
-	// DeleteAttachmentType deletes the attachmentType from the spec
-	DeleteAttachmentType() error
-	// DeleteAttachmentType deletes the attachmentType from the spec
-	DeleteCNIType() error
+// NewFromKubeObject creates a new parser interface
+// It expects a *fn.KubeObject as input representing the serialized yaml file
+func NewFromKubeObject(o *fn.KubeObject) (*Interface, error) {
```

Comment:
In the same spirit as above comment, show we just return the base KubeObjectExt from here?
In fact I am also thinking can the function devs directly invoke kubeobject.NewFromKubeObject[*vlanv1alpha1. Interface]?
That could mean do we really need this library for indirection?

Reply:
The value is you get explicit type checking for the object and we need this in several libraries. This is the reason to do this. otherwise people need to implement this in each function independently again and again.

Reply:
what I did not do so far is have the explicit SeptSpec and SetStatus with their specific type, so this would add even more type awareness.

Reply:
thanks @henderiw . Approved